engine/assets/ui/color_rect.fs

```diff
@@ -1,6 +1,13 @@
 #version 330 core
 
-uniform vec4 color;
+uniform PerElement
```

Comment:
```suggestion
layout(std140) uniform PerElement
```
Without this, there's no guarantee that the fields will be in the order you expect (this might have been what was causing the red tint now that I think of it)